logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetConversationVerificationStatusUseCase.kt

```diff
@@ -0,0 +1,105 @@
+ * [ConversationVerificationStatusResult.Success] with [ConversationVerificationStatus] and [ConversationProtocol] of the conversation
+ * in other cases.
+ */
+interface GetConversationVerificationStatusUseCase {
+    suspend operator fun invoke(conversationId: ConversationId): ConversationVerificationStatusResult
+}
+
+class GetConversationVerificationStatusUseCaseImpl(
+    private val featureSupport: FeatureSupport,
+    private val clientRepository: ClientRepository,
+    private val conversationRepository: ConversationRepository,
+    private val mlsConversationRepository: MLSConversationRepository,
+    kaliumDispatcher: KaliumDispatcher = KaliumDispatcherImpl
+) : GetConversationVerificationStatusUseCase {
+    private val dispatcher = kaliumDispatcher.io
+
+    override suspend fun invoke(conversationId: ConversationId): ConversationVerificationStatusResult = withContext(dispatcher) {
+        if (!featureSupport.isMLSSupported || !clientRepository.hasRegisteredMLSClient().getOrElse(false)) {
+            kaliumLogger.d("Skip getting MLS verification status for conversation, since MLS is not supported.")
+            getConversationProteusVerificationStatus(conversationId)
+        } else {
+            conversationRepository.baseInfoById(conversationId).flatMap { getConversationVerificationStatus(it) }
+        }
+            .getOrElse { ConversationVerificationStatusResult.Failure(it) }
+    }
+
+    private suspend fun getConversationVerificationStatus(
+        conversation: Conversation
+    ): Either<CoreFailure, ConversationVerificationStatusResult> =
+        if (conversation.protocol is Conversation.ProtocolInfo.MLS) {
+            getConversationMLSVerificationStatus(conversation.protocol)
+        } else {
+            getConversationProteusVerificationStatus(conversation.id)
+        }
+
+    private suspend fun getConversationMLSVerificationStatus(protocol: Conversation.ProtocolInfo.MLS) =
+        mlsConversationRepository.getConversationVerificationStatus(protocol.groupId)
+            .map { ConversationVerificationStatusResult.Success(ConversationProtocol.MLS, it) }
+
+    private suspend fun getConversationProteusVerificationStatus(
+        conversationId: ConversationId
+    ): Either<CoreFailure, ConversationVerificationStatusResult> {
+        // TODO
+        return Either.Right(
+            ConversationVerificationStatusResult.Success(
+                ConversationProtocol.PROTEUS,
+                ConversationVerificationStatus.NOT_VERIFIED
+            )
+        )
+    }
+}
+
+sealed class ConversationVerificationStatusResult {
+    data class Success(
+        val protocol: ConversationProtocol,
+        val status: ConversationVerificationStatus
+    ) : ConversationVerificationStatusResult()
+
+    data class Failure(val storageFailure: CoreFailure) : ConversationVerificationStatusResult()
+}
+
+enum class ConversationProtocol {
```

Comment:
why we need this protocol defined here? I think we have it already defined somewhere else.

Reply:
it's needed to determine which protocol is verified and display corresponding icon in UI